spec/unit/interface/action_spec.rb

```diff
@@ -644,4 +644,37 @@ def report(arg)
       expect(subject.get_default_action).to eq(action)
     end
   end
+
+  context "when deprecating a face action" do
+    let :face do
+      Puppet::Interface.new(:foo, '1.0.0') do
+        action :bar do
+          option "--bar"
+          when_invoked do |options| options end
+        end
+      end
+    end
+
+    let :action do face.get_action :bar end
+
+    describe "#deprecate" do
+      it "should set the deprecated value to true" do
+        expect(action.deprecated?).to be_falsey
+        action.deprecate
+        expect(action.deprecated?).to be_truthy
```

Comment:
nit, rspec has special logic for predicate methods so if you do:

```
expect(action).to be_deprecated
```

and it will give a more meaningful error message if the expectation fails. Otherwise you'll get false doesn't equal true.